proto/storage/cluster.proto

```diff
@@ -94,6 +95,56 @@ message AdmissionControllerConfig {
   bool enforce_on_updates = 5;
 }
 
+message ProcessConfig {
+  bool enabled = 1;
+}
+
+message NetworkConnectionConfig {
+  bool enabled = 1;
+  bool aggregate_external = 2;
+}
+
+message NetworkEndpointConfig {
+  bool enabled = 1;
+  bool include_listening_endpoint_processes = 2;
+}
+
+message CollectorFeature {
+  oneof feature {
+    ProcessConfig processes = 1;
+    NetworkConnectionConfig network_connections = 2;
+    NetworkEndpointConfig network_endpoints = 3;
+  }
+}
+
+message CollectorNamespaceConfig {
+  message NamespaceRule {
+    string namespace = 1;
+    storage.MatchType match_type = 2;
+  }
+
+  CollectorFeature feature = 1;
+  repeated NamespaceRule namespace_selection = 2;
+}
+
+// CollectorConfig controls which type of data is reported by collector
+// and processing of that data by collector. For now this configuration
+// controls if processes, network connections, and network endpoints are
+// reported. It also controls if process information is reported with
+// network endpoints and if external IPs are aggregated. External IP
+// aggregation is controlled at the namespace level and in the future
+// other features may also be controlled at the namespace level.
```

Comment:
I think the doc here is too concrete too and should move some to the individual configurations for documentation and their impact. 

```suggestion
// CollectorConfig controls which type of data is reported by collector
// and how it is processed by collector. This configurations are used to fine-tune collector's performance on large scale clusters. 
For example, it controls if process information is reported with network information and if external IPs should be aggregated. 
```

Reply:
Looks good, except "This configurations" should be "These configurations".

Reply:
Done